server/repository/src/db_diesel/name.rs

```diff
@@ -124,15 +124,76 @@ impl<'a> NameRepository<'a> {
 
         let result = final_query.load::<NameAndNameStoreJoin>(&self.connection.connection)?;
 
-        Ok(result.into_iter().map(to_domain).collect())
+        Ok(result.into_iter().map(Name::from_join).collect())
+    }
+
+    pub fn create_filtered_query(store_id: String, filter: Option<NameFilter>) -> BoxedNameQuery {
+        let mut query = name_dsl::name
+            .left_join(
+                name_store_join_dsl::name_store_join.on(name_store_join_dsl::name_id
+                    .eq(name_dsl::id)
+                    .and(name_store_join_dsl::store_id.eq(store_id.clone()))),
+            )
+            .left_join(store_dsl::store)
```

Comment:
```suggestion
            .left_join(store_dsl::store) // if the name found is a store, join the store information
```

Reply:
Hmm, as oppose to store information of store_id parameter ?

Reply:
Hmm, This might be a lack of understanding on my part.
Can a name have a store_id without being a `store`?
I assumed there was a one to <=one relationship between name and store?
Feel free to exclude the comment, I don't think it helps much :)

Reply:
yeah it's one to one, but i wasn't sure about that comment since it's just a join, and the only real need to have this one is to indicate that it's different form `store_id` parameter for this method